optuna/storages/grpc/_server.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime
+import json
+import threading
+
+from optuna.distributions import distribution_to_json
+from optuna.distributions import json_to_distribution
+from optuna.exceptions import DuplicatedStudyError
+from optuna.storages import BaseStorage
+from optuna.storages.grpc._grpc_imports import _imports
+from optuna.study._study_direction import StudyDirection
+from optuna.trial._frozen import FrozenTrial
+from optuna.trial._state import TrialState
+
+
+if _imports.is_successful():
+    from optuna.storages.grpc._grpc_imports import api_pb2
+    from optuna.storages.grpc._grpc_imports import api_pb2_grpc
+    from optuna.storages.grpc._grpc_imports import grpc
+else:
+
+    class api_pb2_grpc:  # type: ignore
+        class StorageServiceServicer:
+            pass
+
+
+DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+
+class OptunaStorageProxyService(api_pb2_grpc.StorageServiceServicer):
+    def __init__(self, storage: BaseStorage) -> None:
+        self._backend = storage
+        self._lock = threading.Lock()
+
+    def CreateNewStudy(
```

Comment:
We use snake_case as standard in Optuna, but should we make an exception and use camelCase for the methods in this class?